src/Tracing/PropagationContext.php

```diff
@@ -201,6 +237,25 @@ private static function parseTraceparentAndBaggage(string $traceparent, string $
             $context->dynamicSamplingContext = $samplingContext;
         }
 
+        // Store the propagated trace sample rand or generate a new one
+        if ($samplingContext->has('sample_rand')) {
+            // TODO check for 1e13 etc.
+            $context->sampleRand = (float) $samplingContext->get('sample_rand');
+        } else {
+            if ($samplingContext->has('sample_rate') && $context->parentSampled !== null) {
+                if ($context->parentSampled === true) {
+                    // [0, rate)
+                    $context->sampleRand = round(mt_rand(0, mt_getrandmax() - 1) / mt_getrandmax() * (float) $samplingContext->get('sample_rate'), 6);
+                } else {
+                    // [rate, 1]
```

Comment:
```suggestion
                    // [rate, 1)
```

1 is an exclusive upper bound